cmd/cluster-agent/api/agent/agent.go

```diff
@@ -149,16 +158,34 @@ func getCheckLatestEvents(w http.ResponseWriter, r *http.Request) {
 }
 
 func getPodMetadata(w http.ResponseWriter, r *http.Request) {
+	if err := apiutil.Validate(w, r); err != nil {
+		return
+	}
 	vars := mux.Vars(r)
+	var err error
+	var slcB []byte
 	nodeName := vars["nodeName"]
 	podName := vars["podName"]
-	svcList := as.GetPodServiceNames(nodeName, podName)
 
-	slcB, err := json.Marshal(svcList)
+	if podName == "*" {
```

Comment:
This is not common, let's say I have a route to get a pods like `/api/pod/<str: podName>`
If you want to get all pods, you should hit the `/` of the route like:
`/api/pod/` instead of creating special chars for that (`*`).

Reply:
Agreed it's weird, If you want to follow REST guidelines `/api/pods` would get all the pods and `/api/pods/<pod-id>` whould get a specific pod 📖

Reply:
Thanks !